### PR TITLE
Fix SelectComponent ctor not setting component type

### PR DESCRIPTION
**Pull request by VelvetThePanda**
I forgot to add `: this()` to SelectComponent

This also fixes a bug in interactivity where it would say no dropdowns existed with the specified Id

### CODE REVIEW

File: DSharpPlus/Entities/Interaction/Components/DiscordSelectComponent.cs [hunk 1]
```diff
@@ -33,12 +33,6 @@ namespace DSharpPlus.Entities
     /// </summary>
     public sealed class DiscordSelectComponent : DiscordComponent
     {
-        /// <summary>
-        /// The custom Id of this component.
-        /// </summary>
-        [JsonProperty("custom_id", NullValueHandling = NullValueHandling.Ignore)]
-        public new string CustomId { get; set; }
```
Comment (Lulalaby) [inline DSharpPlus/Entities/Interaction/Components/DiscordSelectComponent.cs:40]: Is this because the custom id is already defined in DiscordComponent?

Comment (VelvetThePanda) [reply DSharpPlus/Entities/Interaction/Components/DiscordSelectComponent.cs:40]: Yes.

File: DSharpPlus/Entities/Interaction/Components/DiscordSelectComponent.cs [hunk 2]
```diff
@@ -68,7 +62,7 @@ public DiscordSelectComponent()
             this.Type = ComponentType.Select;
         }
 
-        public DiscordSelectComponent(string customId, IEnumerable<DiscordSelectComponentOption> options, string placeholder = null, int minOptions = 1, int maxOptions = 1)
+        public DiscordSelectComponent(string customId, string placeholder = null, IEnumerable<DiscordSelectComponentOption> options = default, int minOptions = 1, int maxOptions = 1) : this()
```
Comment (Neuheit) [inline DSharpPlus/Entities/Interaction/Components/DiscordSelectComponent.cs:65]: I don't think options are an optional parameter

Comment (VelvetThePanda) [reply DSharpPlus/Entities/Interaction/Components/DiscordSelectComponent.cs]: Should be fixed

File: DSharpPlus/Entities/Interaction/Components/DiscordSelectComponent.cs [hunk 3]
```diff
@@ -33,42 +33,36 @@ namespace DSharpPlus.Entities
...
         [JsonProperty("min_values", NullValueHandling = NullValueHandling.Ignore)]
-        public int? MinimumSelectedValues { get; set; }
+        public int? MinimumSelectedValues { get; internal set; }
 
         /// <summary>
         /// The maximum amount of options that can be selected. Must be greater than zero and or equal to <see cref="MinimumSelectedValues"/>. Defaults to 1.
         /// </summary>
         [JsonProperty("max_values", NullValueHandling = NullValueHandling.Ignore)]
-        public int? MaximumSelectedValues { get; set; }
+        public int? MaximumSelectedValues { get; internal set; }
```
Comment (Lulalaby) [inline DSharpPlus/Entities/Interaction/Components/DiscordSelectComponent.cs:58]:
```suggestion
        public int? MaximumSelectedValues { get; internal set; }
        
        /// <summary>
        /// Whether this select can be used.
        /// </summary>
        [JsonProperty("disabled", NullValueHandling = NullValueHandling.Ignore)]
        public bool Disabled { get; internal set; }
```

File: DSharpPlus/Entities/Interaction/Components/DiscordSelectComponent.cs [hunk 4]
```diff
@@ -33,42 +33,36 @@ namespace DSharpPlus.Entities
...
 
         public DiscordSelectComponent()
         {
             this.Type = ComponentType.Select;
         }
 
-        public DiscordSelectComponent(string customId, IEnumerable<DiscordSelectComponentOption> options, string placeholder = null, int minOptions = 1, int maxOptions = 1)
+        public DiscordSelectComponent(string customId, string? placeholder, IEnumerable<DiscordSelectComponentOption> options, int minOptions = 1, int maxOptions = 1) : this()
```
Comment (Lulalaby) [inline DSharpPlus/Entities/Interaction/Components/DiscordSelectComponent.cs:65]:
```suggestion
        public DiscordSelectComponent(string customId, string? placeholder, IEnumerable<DiscordSelectComponentOption> options, int minOptions = 1, int maxOptions = 1,bool disabled = false) : this()
```

File: DSharpPlus/Entities/Interaction/Components/DiscordSelectComponent.cs [hunk 5]
```diff
@@ -33,42 +33,36 @@ namespace DSharpPlus.Entities
...
         {
             this.CustomId = customId;
             this.Options = options.ToArray();
```
Comment (Lulalaby) [inline DSharpPlus/Entities/Interaction/Components/DiscordSelectComponent.cs:68]:
```suggestion
            this.Disabled = disabled;
            this.Options = options.ToArray();
```

File: DSharpPlus/Entities/Interaction/Components/DiscordSelectComponent.cs (hunk 3 above)
Comment (VelvetThePanda) [reply DSharpPlus/Entities/Interaction/Components/DiscordSelectComponent.cs:58]: [Select menus can't be disabled](https://discord.dev/interactions/message-components#select-menu-object-select-menu-structure)

File: DSharpPlus/Entities/Interaction/Components/DiscordSelectComponent.cs (hunk 4 above)
Comment (VelvetThePanda) [reply DSharpPlus/Entities/Interaction/Components/DiscordSelectComponent.cs:65]: See above comment

File: DSharpPlus/Entities/Interaction/Components/DiscordSelectComponent.cs (hunk 3 above)
Comment (Lulalaby) [reply DSharpPlus/Entities/Interaction/Components/DiscordSelectComponent.cs:58]: They can

Comment (Lulalaby) [reply DSharpPlus/Entities/Interaction/Components/DiscordSelectComponent.cs:58]: https://github.com/discord/discord-api-docs/pull/3209

**Pull request merged**